shapely/_geometry_helpers.pyx

```diff
@@ -118,24 +122,22 @@ def simple_geometries_1d(object coordinates, object indices, int geometry_type,
                         f"Index {geom_idx} is missing from the input indices."
                     )
 
-            # check if we need to close a linearring
-            if geometry_type == 2:
-                ring_closure = 0
-                if geom_size == 3:
-                    ring_closure = 1
-                else:
-                    for coord_idx in range(dims):
-                        if coord_view[idx, coord_idx] != coord_view[idx + geom_size - 1, coord_idx]:
-                            ring_closure = 1
-                            break
-                # check the resulting size to prevent invalid rings
-                if geom_size + ring_closure < 4:
+            errstate = PyGEOS_CoordSeq_FromBuffer(geos_handle, &coord_view[idx, 0], geom_size, dims, is_ring, handle_nans, &seq)
+            if errstate == 9:  # PGERR_NAN_COORD (in shapely/src/geos.h)
```

Comment:
One thing I am wondering is if we should "expose" those exact numbers from our C code to cython. As long as we are only using this ourselves, that is fine of course (we can always update the above code if that number would change). 

But if we want to make it official that third party packages can use the C/cython API, we could also change `PyGEOS_CoordSeq_FromBuffer` to catch the return value of `coordseq_from_buffer` and translate it into something more generic (0/1/2 or 0/1/-1)